inbm/dispatcher-agent/dispatcher/ota_parser.py

```diff
@@ -55,7 +55,7 @@ def parse(self, resource: Dict, kwargs: Dict, parsed: XmlHandler) -> Dict[str, A
         self._password = resource.get('password', None)
         self._signature = resource.get('signature', None)
         if self._signature:
-            self._hash_algorithm = int(resource.get('sigversion', 384))
+            self._hash_algorithm = int(resource.get('sigversion', DEFAULT_HASH_ALGORITHM))
```

Comment:
**Suggestion:** Ensure that the 'sigversion' key exists in the 'resource' dictionary before attempting to cast its value to an integer. This prevents a potential KeyError if 'sigversion' is not provided. [possible issue]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if self._signature:
            sigversion = resource.get('sigversion')
            if sigversion is not None:
                self._hash_algorithm = int(sigversion)
            else:
                self._hash_algorithm = DEFAULT_HASH_ALGORITHM
```